Add AWS_CRT_BUILD_USE_SYSTEM_LIBCRYPTO (#456)
awslabs/aws-crt-python: README.md

This library is licensed under the Apache 2.0 License.

## Minimum Requirements:

*   Python 3.7+

## Installation

To install from pip:

```bash
python3 -m pip install awscrt
```

To install from Github:

```bash
git clone https://github.com/awslabs/aws-crt-python.git
cd aws-crt-python
git submodule update --init
python3 -m pip install .
```

To use from your Python application, declare `awscrt` as a dependency in your `setup.py` file.

### OpenSSL and LibCrypto (Unix only)

aws-crt-python does not use OpenSSL for TLS.
On Apple and Windows devices, the OS's default TLS library is used.
On Unix devices, [s2n-tls](https://github.com/aws/s2n-tls) is used.
But s2n-tls uses libcrypto, the cryptography math library bundled with OpenSSL.

> You can ignore all this on Windows and Apple platforms, where aws-crt-python
> uses the OS's default libraries for TLS and cryptography math.
> 

Reviewer: Oops, this sentence is in here twice. Once here, and again at the end of the section. let's remove this one. I edit things to death, and sometimes leave in ghosts of old edits by accident
```suggestion
```

To simplify installation, aws-crt-python has its own copy of libcrypto.
This lets you install a wheel from PyPI without having OpenSSL installed.
Unix wheels on PyPI come with libcrypto statically compiled in.
Code to build libcrypto comes from [AWS-LC](https://github.com/aws/aws-lc).
AWS-LC's code is included in the PyPI source package, 
and the git repository includes it as a submodule.

If you need aws-crt-python to use the libcrypto included on your system, 
set environment variable `AWS_CRT_BUILD_USE_SYSTEM_LIBCRYPTO=1` while building from source:

```sh
AWS_CRT_BUILD_USE_SYSTEM_LIBCRYPTO=1 python3 -m pip install --no-binary :all: --verbose awscrt
```

> ( `--no-binary :all:` ensures you do not use the wheel from PyPI always uses AWS-LC)

Reviewer: another ghost of edits past
```suggestion
( `--no-binary :all:` ensures you do not use the precompiled wheel from PyPI)
```

```
static: certificate has an existing certificate-key pair that was previously imported into the Keychain. Using key from Keychain instead of the one provided.
```

> ### OpenSSL and LibCrypto (Unix only)

Reviewer: Move this up above the `Mac-Only TLS Behavior` section

This openssl stuff is relevent to building/installation, it should be close to the existing build/install steps

> aws-crt-python does not use OpenSSL for TLS.
> On Apple and Windows devices, the OS's default TLS library is used.
> On Unix devices, [s2n-tls](https://github.com/aws/s2n-tls) is used.
> But s2n-tls uses libcrypto, the cryptography math library bundled with OpenSSL.
> You can ignore all this on Windows and Apple platforms, where aws-crt-python
> uses the OS's default libraries for TLS and cryptography math.
> To simplify installation, aws-crt-python has its own copy of libcrypto.
> This lets you install a wheel from PyPI without having OpenSSL installed.
> Unix wheels on PyPI come with libcrypto statically compiled in.
> Code to build libcrypto comes from [AWS-LC](https://github.com/aws/aws-lc).
> AWS-LC's code is included in the PyPI source package, 
> and the git repository includes it as a submodule.
> If you need aws-crt-python to use the libcrypto included on your system, 
> set environment variable `AWS_CRT_BUILD_USE_SYSTEM_LIBCRYPTO=1` while building from source:

Reviewer: paragraph breaks
```suggestion
aws-crt-python does not use OpenSSL for TLS.
On Apple and Windows devices, the OS's default TLS library is used.
On Unix devices, [s2n-tls](https://github.com/aws/s2n-tls) is used.
But s2n-tls uses libcrypto, the cryptography math library bundled with OpenSSL.

You can ignore all this on Windows and Apple platforms, where aws-crt-python
uses the OS's default libraries for TLS and cryptography math.

To simplify installation, aws-crt-python has its own copy of libcrypto.
This lets you install a wheel from PyPI without having OpenSSL installed.
Unix wheels on PyPI come with libcrypto statically compiled in.
Code to build libcrypto comes from [AWS-LC](https://github.com/aws/aws-lc).
AWS-LC's code is included in the PyPI source package, 
and the git repository includes it as a submodule.

If you need aws-crt-python to use the libcrypto included on your system, 
set environment variable `AWS_CRT_BUILD_USE_SYSTEM_LIBCRYPTO=1` while building from source:
```

```sh
AWS_CRT_BUILD_USE_SYSTEM_LIBCRYPTO=1 python3 -m pip install --no-binary :all: --verbose awscrt
```

> 
> ( `--no-binary :all:` ensures you do not use the wheel from PyPI always uses AWS-LC)
> You can ignore all this on Windows and Apple platforms, where aws-crt-python
> uses the OS's default libraries for TLS and cryptography math.

Reviewer: paragraph breaks

```suggestion
( `--no-binary :all:` ensures you do not use the wheel from PyPI always uses AWS-LC)

You can ignore all this on Windows and Apple platforms, where aws-crt-python
uses the OS's default libraries for TLS and cryptography math.
```
